pkg/controller/deliverable/reconciler.go

```diff
@@ -109,16 +120,26 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			err = controller.NewUnhandledError(err)
 		}
 	} else {
+		if log.V(logger.DEBUG).Enabled() {
```

Comment:
tbh, this loop is soo cheap that I'd go without the branching .. but happy to hear from other folks too

Reply:
it's not as cheap as fmt.Sprintf. Objects cost a little to serialize, and a _lot_ to serialize if logging dev_mode is enabled. I'd happily wait for profiling efforts too though